test/sort.jl

```diff
@@ -30,4 +39,88 @@ using StaticArrays, Test
         @test sortperm(SA[1, 1, 1, 0]) == SA[4, 1, 2, 3]
     end
 
-end
+    @testset "NaNs" begin
+        # Return an SVector with floats and NaNs that have random sign and payload bits.
+        @generated function floats_randnans(::Type{SVector{N, T}}, p) where {N, T}
```

Comment:
Okay, I've made these changes. I use a very similar generated function (`floats_nans`) in `bench_sort.jl` to do the same unrolling. and I haven't changed that function because I'm unable to achieve the same speed and non-allocation with a non-generated version. (Doesn't matter here though since speed is not an issue.)

Reply:
Actually, I just realized that `ntuple(f, Val(N))` with `@inline f` can be used to force the same unrolling even for arbitrarily large `N`s, so I've now replaced both generated functions with equivalent non-generated functions.